src/e2e-test/java/io/cdap/plugin/common/stepsdesign/BQValidationExistingTables.java

```diff
@@ -0,0 +1,116 @@
+   * @param fileName The name of the JSON file containing the expected data.
+   * @return True if the actual data matches the expected data, false otherwise.
+   */
+  public static boolean validateActualDataToExpectedData(String table, String fileName) throws IOException,
+    InterruptedException, URISyntaxException {
+    Map<String, JsonObject> bigQueryMap = new HashMap<>();
+    Map<String, JsonObject> fileMap = new HashMap<>();
+    Path  bqExpectedFilePath = Paths.get(BQValidationExistingTables.class.getResource("/" + fileName).toURI());
+
+    getBigQueryTableData(table, bigQueryMap);
+    getFileData(bqExpectedFilePath.toString(), fileMap);
+    boolean isMatched = bigQueryMap.equals(fileMap);
+    return isMatched;
+  }
+
+  /**
+   * Reads a JSON file line by line and populates a map with JSON objects using a specified ID key.
+   *@param fileName The path to the JSON file to be read.
+   * @param fileMap  A map where the extracted JSON objects will be stored with their ID values as keys.
+   */
+
+  public static void getFileData(String fileName, Map<String, JsonObject> fileMap) {
+    try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        JsonObject json = gson.fromJson(line, JsonObject.class);
+        String idKey = getIdKey(json);
+        if (idKey != null) {
+          JsonElement idElement = json.get(idKey);
+          if (idElement.isJsonPrimitive()) {
+            String idValue = idElement.getAsString();
+            fileMap.put(idValue, json);
+          }
+        } else {
+          Log.error("ID key not found");
+        }
+      }
+    } catch (IOException e) {
+      Log.error("Error reading the file: " + e.getMessage());
+    }
+  }
+
+  private static void getBigQueryTableData(String targetTable, Map<String, JsonObject> bigQueryMap)
+    throws IOException, InterruptedException {
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + targetTable + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+
+    for (FieldValueList row : result.iterateAll()) {
+      JsonObject json = gson.fromJson(row.get(0).getStringValue(), JsonObject.class);
+      String idKey = getIdKey(json); // Get the actual ID key from the JSON object
+      if (idKey != null) {
+        JsonElement idElement = json.get(idKey);
+        if (idElement.isJsonPrimitive()) {
+          String id = idElement.getAsString();
+          bigQueryMap.put(id, json);
+        } else {
+          Log.error("Data Mismatched");
+        }
+      }
+    }
+  }
+
+  /**
+   * Retrieves the key for the ID element in the provided JSON object.
+   *
+   * @param json The JSON object to search for the ID key.
+   */
+  private static String getIdKey(JsonObject json) {
```

Comment:
Can we remove this function getIdKey and use only id as a unique identifier in above functions instead of using this logic if not neccessary.